shopfloor/actions/stock.py

```diff
@@ -11,10 +13,14 @@ class StockAction(Component):
     _usage = "stock"
 
     def mark_move_line_as_picked(
-        self, move_lines, quantity=None, package=None, user=None
+        self, move_lines, quantity=None, package=None, user=None, check_user=False
     ):
         """Set the qty_done and extract lines in new order"""
         user = user or self.env.user
+        if check_user:
+            picking_users = move_lines.mapped("picking_id.user_id")
+            if not all(pick_user == user for pick_user in picking_users):
+                raise UserError("Someone already working on this")
```

Comment:
```suggestion
                raise UserError("Someone is already working on these transfers")
```

Reply:
{user} has already processed this transfer

Reply:
> {user} has already processed this transfer

But there could be multiple users and transfers, no ?

Reply:
most of the time you process one move line and it could be processed by one user